post-types/ckan-backend-local-harvester.php

```diff
@@ -342,4 +341,147 @@ public static function get_source_type_form_field_options() {
+
+	/**
+	 * Generates selectbox to filter by harvest-status
+	 *
+	 * @param bool $disable_floating Disable floating of the selectbox which is default in WordPress.
+	 */
+	public static function print_harvest_status_filter( $disable_floating = false ) {
+		$harvest_states = [
+			'errored' => __( 'Errored', 'ogdch-backend' ),
+			'updated' => __( 'Updated', 'ogdch-backend' ),
+			'added' => __( 'Added', 'ogdch-backend' ),
+			'deleted' => __( 'Deleted', 'ogdch-backend' ),
+		];
+		?>
+		<select name="harvest_status_filter" <?php echo ($disable_floating) ? 'style="float: none;"' : ''; ?>>
+			<option value=""><?php esc_attr_e( 'All harvest states', 'ogdch-backend' ); ?></option>
+			<?php
+			$harvest_status_filter   = '';
+			if ( isset( $_GET['harvest_status_filter'] ) ) {
+				$harvest_status_filter = sanitize_text_field( $_GET['harvest_status_filter'] );
+			}
+
+			foreach ( $harvest_states as $harvest_status_key => $harvest_status_value ) {
+				printf(
+					'<option value="%s"%s>%s</option>',
+					esc_attr( $harvest_status_key ),
+					esc_attr( ( $harvest_status_key === $harvest_status_filter ) ? ' selected="selected"' : '' ),
+					esc_attr( $harvest_status_value )
+				);
+			}
+			?>
+		</select>
+		<?php
+	}
+
+	/**
+	 * Retrieves all harvest sources from the CKAN API.
+	 *
+	 * @return array List of all harvest sources.
+	 */
+	protected static function get_harvest_sources() {
+		if ( empty( self::$harvest_sources ) ) {
+			$endpoint = CKAN_API_ENDPOINT . 'harvest_source_list';
+			$data = array( 'return_last_job_status' => true );
+
+			$response = Ckan_Backend_Helper::do_api_request( $endpoint, $data );
+			$errors = Ckan_Backend_Helper::check_response_for_errors( $response );
+
+			if ( 0 === count( $errors ) ) {
+				self::$harvest_sources = $response['result'];
+			} else {
+				Ckan_Backend_Helper::print_error_messages( $errors );
+			}
+		}
+		return self::$harvest_sources;
+	}
+
+	/**
+	 * Returns an Array of keys 'errored', 'updaded', 'added', 'deleted', which contain a list of harvester-ids each.
+	 *
+	 * @return array Array of harvester-job-status.
+	 */
+	protected static function get_harvest_ids_by_status() {
+		if ( empty( self::$harvest_ids_by_states ) ) {
+			$harvest_sources = self::get_harvest_sources();
+			$harvest_sources_by_states = [
+				'errored'	=> array(),
+				'updated'	=> array(),
+				'added'		=> array(),
+				'deleted'	=> array(),
+			];
+
+			foreach ( $harvest_sources as $harvest_source ) {
+				$status = $harvest_source['last_job_status']['stats'];
+
+				if ( $status['errored'] > 0 ) {
+					$harvest_sources_by_states['errored'][] = $harvest_source['id'];
+				} elseif ( $status['updated'] > 0 ) {
+					$harvest_sources_by_states['updated'][] = $harvest_source['id'];
+				} elseif ( $status['added'] > 0 ) {
+					$harvest_sources_by_states['added'][] = $harvest_source['id'];
+				} elseif ( $status['deleted'] > 0 ) {
+					$harvest_sources_by_states['deleted'][] = $harvest_source['id'];
+				}
```

Comment:
The possible values of status are:

* Errored (if `$harvest_source['last_job_status']['stats']['errored']` > 0)
* Finished (if `$harvest_source['last_job_status']['status']` == Finished)
* Running (if `$harvest_source['last_job_status']['status']` == Running)
* New (if `$harvest_source['last_job_status']['status']` == New)